feature-pitches/010-compute-sessions/compute-sessions.md

```diff
@@ -0,0 +1,220 @@
+
+### User Flows
+
+#### User Flow: Add a Session to a Project
+
+![breadboard-add-session-pt-1.jpg](breadboard-add-session-pt-1.jpg)
+
+continuing to the right...
+
+![breadboard-add-session-pt-2.jpg](breadboard-add-session-pt-2.jpg)
+
+A user has 4 options for adding a session to their project. Only the first 2 are considered critical for this build. These options are listed in order of priority, so if this build doesn’t get to implementing all of them, that’s ok.
+
+##### Priority 1: Select from the RenkuLab Default Environments
+
+- These are built and maintained by us
+- **At a minimum, 1 default environment must be created as part of the build!** (a Jupyter Python environment)
+- These images are pre-built, so launching  a session from one is fast (the user doesn’t have to wait for an image build)
+- [nice to have] The default environments should be configured via the admin panel
+
+##### Priority 2: Use an external image
+
+- There are a whole bunch of restrictions for what must be in the image for it to work on RenkuLab. We will have to document these limitations for users.
+  - We *think* that any image coming from the Jupyter stack should work
+  - ... Or at least Renku base images
+- Only publicly accessible images are supported
+
+##### Priority 3 [nice to have]: Configure a Session from a Code repo
+
+- This uses the already existing CI image build pipeline
+- We do not automatically update the image that is used for the session when new commits are pushed and new images are built. It is up to the user if they want to update which image is built
+- [nice-to-have]: The user is notified when a new image is available and invited to change the image their session uses if they wish.
+
+##### Priority 4: [for the future/out of scope] Search & select from preexisting RenkuLab environments
+
+#### User Flow: Start Session
+
+![breadboard-start-session.jpg](breadboard-start-session.jpg)
+
+#### User Flow: Configure Session [nice to have]
+
+![breadboard-configure-session.jpg](breadboard-configure-session.jpg)
+
+#### User Flow: Publish Session [for the future/out of scope]
+
+![breadboard-publish-session.jpg](breadboard-publish-session.jpg)
+
+#### User Flow: Update a Repo-Based Session [nice to have]
+
+![breadboard-update-repo-based-session.jpg](breadboard-update-repo-based-session.jpg)
+
+### Things not communicated via the User Flows
+
+#### Starting Assumptions
+
+- In this build, we will not change the project GitLab CI-based image build process (for 'Configure a Session from a Code repo')
+
+#### Key Points
+
+- When a user creates a new project, the project has no sessions until a user adds one.
+  - [nice to have] Provide shortcuts for adding a session from the RenkuLab default environments
+- A user can create as many sessions as they want in a project.
+- A user can run 1 session per environment per project (similar to our restriction now of 1 session per user per project)
```

Comment:
So we need to actually prevent the user from creating multiple sessions from the same environment - should this be mentioned in the user flow?